### ISSUE TITLE
The extension package that you are trying to install has a different set of tables and table extensions than the extension package that has been previously synchronized. 

### ISSUE THREAD

**Issue by fvet** [microsoft/AL#7364]
Please include the following with each issue:

**1. Describe the bug**

Since the latest AL v11 vsix, I'm no longer able to publish against a local Docker sandbox without increasing the app's version number. An error 'The extension package that you are trying to install has a different set of tables and table extensions than the extension package that has been previously synchronized. ' is shown when publishing from VS Code.

**2. To Reproduce**

I can't find an actual repro, but here's the behavior I encounter 'now and then'....

- Open a VS Code
- Create a new project
- Set the app.json to 1.0
- Publish the app from VS Code to a local dev environment (Sandbox **docker**)
- Perform some code changes (could be adding a field to a table, ...)
- Publish the app from VS Code to the same local dev environment (Sandbox **docker**)
- An error 'The extension package that you are trying to install has a different set of tables and table extensions than the extension package that has been previously synchronized. ' is shown.
- I'm forced to increase the app's version no., which is not wanted.

**3. Expected behavior**

- Same behavior as in v10, simply allow to publish the app without increasing the version no.

**4. Actual behavior**

- I'm forced to increase the app's version no., which is not wanted.

`[2023-04-06 15:53:12.50] The request for path /BC/dev/apps?tenant=default&SchemaUpdateMode=forcesync&ForceUpgrade=true&DependencyPublishingOption=default failed with code UnprocessableEntity. Reason: The extension package for 'Dynavision Excise Duties by Dynavision 22.0.0.1' that you are trying to install has a different set of tables and table extensions than the extension package that has been previously synchronized. You should increase the version of the extension and restart the deployment process or force synchronize the current extension package.`

![image](https://user-images.githubusercontent.com/25268332/230399265-b2e8e6fb-2767-43b1-a60d-6b62c9e818bc.png)

**5. Versions:**

- AL Language: 11.0.779295
- Visual Studio Code:
- Business Central:
- List of Visual Studio Code extensions that you have installed:

**Comment by thloke** [comment]
I'm pretty sure this is caused by a regression on the server, instead of a regression in the VSIX. It's probably not much consolation though, since it's not like you can downgrade to v10 of the VSIX to publish against BC 22. You can confirm this by publishing to an older (v21) docker, the behaviour should be as expected.

Server issue or not, I'll still raise this to be triaged by the team though, it's a usability issue having to increase the versions each time. Sort of defeats the purpose of DEV extensions.

**Comment by pri-kise** [comment]
I have the same problem targeting the release BC22.0 DE Version.
If I change the schemaupdatemode in the launch.json to ForceSync then I'm able to publish with same version number again.

**Comment by TheDenSter** [comment]
This is SO annoying... I deploy very often when I'm developing, like every 5-10 minutes. Having to increase the version number every time is very distracting and in my opinion unreasonable. Turning on ForceSync is a bad workaround because I want to be warned/stopped when I really make a destructive change. 

Can we please make this a setting? At least for working in local containers?

**Comment by qutreson** [comment]
You are correct. You should be able to do additive changes to your table (adding fields or keys) using Synchronize without changing the version number. We have found how the regression was introduced and we are now working on a fix for it. Thank you for raising the issue.

**Comment by fvet** [comment]
> @fvet - Which schema update mode are you using in your launch.json? The screenshot seems to point toward ForceSync, but asking for confirmation. It should work for ForceSync and Recreate, but not for Synchronize.
> 
> Which version of Business Central are you targeting? Can you share the exact version from the help pane?

Sorry for the late reply:

- Here's the BC version :` Version: BE Business Central 22.0 (Platform 22.0.55014.0 + Application 22.0.54157.55041)`
- Launch.json has `ForceSync` as `SchemaUpdateMode`

![image](https://user-images.githubusercontent.com/25268332/232035624-1fd02cf0-9d0a-4490-9dfb-2e3442e6aa67.png)

**Comment by JesperSchulz** [comment]
The fix for this issue has been checked in to the master branch. It will be available in the bcinsider.azurecr.io/bcsandbox-master Docker image starting from platform build number 23.0.10599.0 and VS Code Extension Version 12.0.790342. 
 
 If you don’t have access to these images you need to become part of the Ready2Go program: aka.ms/readytogo 
 
 For more details on code branches and docker images please read: 
 https://blogs.msdn.microsoft.com/nav/2018/05/03/al-developer-previews-multiple-releases-and-github/ 
 https://freddysblog.com/2020/06/25/working-with-artifacts/

**Comment by fvet** [comment]
@JesperSchulz  I hope this will be included in **22**.x (nextMinor) and in VS Code extension **11**.x as a hotfix as well?

**Comment by CarlosAndreuLlaneras** [comment]
Hi @JesperSchulz, 
We have deployed new dockers (BC22.1 and BC23.0) and updated VSC AL ext. (11.0.793701) and still facing the same issue. 
Can you confirm that the fix has already been included in the artifacts for BC 22.1 and 23.0?
We are deploying the new dockers by downloading the artifacts from:
https://bcinsider.azureedge.net/sandbox/22.1.56280.0/es
https://bcinsider.azureedge.net/sandbox/23.0.10701.0/es

What am I missing?

Thanks

**Comment by clasalal** [comment]
Indeed, the issue persists in the latest bc images if you have the 'forceUpgrade' parameter set to true in the launch.json file. If this parameter is removed, the error no longer occurs. Until now, there was no problem having this parameter set to true, so something must have changed. Can you please confirm? Thank you.

**Comment by JesperSchulz** [comment]
@thloke, @qutreson, could you look into the above comment?

**Comment by qutreson** [comment]
@clasalal can you log a new issue regarding the 'forceUpgrade' scenario and we will look into it
